File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    var answer = [];
    let singer = new Array(k);
    
    for (let i=0; i<score.length; i++){
        if(i<k){
            singer.push(score[i]);
            singer.sort((a, b) => a-b);
            answer.push(singer[0]);
        } else{
            if(score[i]>singer[0]){
                singer[0] = score[i];
                singer.sort((a, b) => a-b);
            }
            answer.push(singer[0])
        }
    }
    return answer;
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = []
    singer = []
    for i in range(len(score)):
        if i < k:
            singer.append(score[i]) 
            singer.sort()
            answer.append(singer[0])
        else:
            if score[i] > singer[0]:
                singer[0] = score[i]
                singer.sort()
        
        if i >= k:
            answer.append(singer[0])
            
    return answer